### PR TITLE
Fix max_file_frames and max_total_frames functionality

### LINKED ISSUE #1262: Image series import doesn't respect the max number of frames requested
@joelvbernier Will add test case.

### PR DESCRIPTION
**Pull request by bnmajor**
Fixes #1262

This PR fixes the bugs for now but I will be doing some refactoring of this shared state code (omegas, empty/max frames, etc) as well as some class refactoring to prevent these problems from bubbling up again down the road. Follow up PR with those changes to come.

### CODE REVIEW

**Comment by bnmajor** [comment]
The PEP 8 issues are from 3 instances of walrus operators.

File: hexrd/ui/image_stack_dialog.py [hunk 1]
```diff
@@ -359,37 +361,44 @@ def get_omega_values(self, num_files):
...
         else:
             # user did not select file or enter wedges
             delta = MAXIMUM_OMEGA_RANGE / num_files
             omega = np.linspace(
                 [0, 0 + delta],
                 [MAXIMUM_OMEGA_RANGE - delta, MAXIMUM_OMEGA_RANGE],
-                num_files)
+                num_files).astype('uint16')
+            omega = [[b, e, self.frames_per_image] for [b, e] in omega]
+            if max_total := self.state['max_total_frames']:
```
Comment (psavery) [inline hexrd/ui/image_stack_dialog.py:376]: So this `if` statement is just checking that the `max_total` is not set to zero, right?

**Comment by bnmajor** [comment]
@cjh1 This should fix the bug that you were seeing

**Review by cjh1 — APPROVED** [review]
LGTM

**Pull request merged**